README.md

```diff
@@ -200,6 +200,28 @@ Then you can simply point the converted model path in the config file by changin
 
+To finetune the pretrained maskRCNN to train fasterRCNN or maskRCNN on your own dataset, four steps are needed: <br />
+
+**step 1**. Run the demo.py to test your package is installed successfully. Meanwhile, you can save the pretrained model, which has been converted from pkl format to pth format, which can be loaded by a pytorch model. The model is a data of COCODEmo object in the demo, so you can save the weights of the pretrained model as follow: 
+```python
+    pretrained_model_path = "./pretrained_model/maskRCNN.pth"
+    check_point = {'state': coco_demo.model.state_dict()}
+    torch.save(check_point, pretrained_model_path)
+```
+This way, no need to look into [_load_file fun](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/utils/checkpoint.py#L117). BTW, in case you want to look into this: the [state_dict](https://github.com/pytorch/pytorch/blob/828cb18fa35c7c132ab920de1c0dc6d859f152d6/torch/nn/modules/module.py#L602) method of a Module object saves the weights in an OrderedDict. <br />
+
+**step 2**. Create a fastercnn or maskRCNN model using a yaml file in the configs folder. When doing this, you need to call the cfg.merge_from_list method to change the [_C.MODEL.ROI_BOX_HEAD.NUM_CLASSES =81 ](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/config/defaults.py#L182) to the number of classes in your dataset (background does not count in this implementation).
+
+**step 3**. Two classes are needed. The 1st class to create fetches one image and its corresponding bbox and label without bundling the bbox and label in a BoxList object. This class is a counterpart of the parent class of the [COCODataset](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/data/datasets/coco.py#L9).  In each iteration, this class generates a tuple (img, bbox, label), in which img is generated by Image.open(...).convert('RGB'); bbox is a 2d numpy array and label is 1d array (mask information if it exists). The 2nd class inherits from the 1st class and the major responsibility of this class is to wrap the truths of an image in a BoxList object, similar to this [line](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/data/datasets/coco.py#L43).
+
+**step 4**. For a pytorch pretrained model, the weights are storted in a dictionary, in which the key is the name of a layer. So you can exclude layers that do not exist or you want train from scratch as follows:
+```python
+pretrained_model = torch.load(checkpoint_path)['state']
+model_dict = self.model.state_dict()
+pretrained_dict = {k: v for k, v in pretrained_model.items() if k in model_dict and "roi_heads" not in k}
+model_dict.update(pretrained_dict)
```

Comment:
This assumes that we are modifying the training code, right?
Can we maybe export the file here so that the user can just point to the new weights file in his config?

Reply:
Yes. I think we can do surgery in the load_state_dict function as it seems that checking which keys in the [loaded_state_dict] (https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/utils/model_serialization.py#L71) does not affect other parts in the repo. The code would be something as follows:
model_dict = model.state_dict()
loaded_state_dict = {k: v for k, v in loaded_state_dict.items() if k in model_dict and "roi_heads" not in k}
model_dict.update(loaded_state_dict)

If so, no need to go through the 4 steps. Then I can close this pull and send a PR to edit the [load_state_dict method] (https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/utils/model_serialization.py#L71). How do you think?

Reply:
I think that `load_state_dict` already supports missing keys, thanks to the `align_and_update_state_dicts` function https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/utils/model_serialization.py#L77

Thoughts?